static-assets/components/cstudio-forms/controls/node-selector.js

```diff
@@ -396,38 +397,36 @@ YAHOO.extend(CStudioForms.Controls.NodeSelector, CStudioForms.CStudioFormField,
             '<button class="fa fa-trash node-selector-item-icon" title="Delete" aria-label="Delete" role="button"></button>'
           );
           const isEditable = this.allowEdit && (isComponent || craftercms.utils.content.isEditableAsset(item.key));
-          if (isEditable) {
-            if (isComponent || !this.readonly) {
```

Comment:
See how `isComponent` is already checked in the `isEditable` validation, and readOnly is already validated in `hasEditAction`. That `isComponent` validation is not valid anymore because basically just because it was a component it was allowing the edit action. Now we consider the permissions

I'll add comments for it to be clearer.
There are 3 variables related with edition:
- allowEdit: the datasource has edit function (`datasource.edit` exists)
- isEditable: the item can be edited by studio (e.g. is a component, or a text file). A non-editable item is for example an image.
- hasEditAction: there's where we check for the edit permission, but if the item is an inline component, then we check for the readonly property.